webapp/tests/karma/ts/services/uhc-settings.service.spec.ts

```diff
@@ -0,0 +1,82 @@
+import { TestBed } from '@angular/core/testing';
+import { expect } from 'chai';
+import { UHCSettingsService } from '@mm-services/uhc-settings.service';
+
+describe('UHCSettings service', () => {
+  let service: UHCSettingsService;
+
+  beforeEach(() => {
+    service = TestBed.inject(UHCSettingsService);
+  });
+
+  describe('getMonthStartDate', () => {
+    it('should return falsy for invalid input', () => {
+      expect(!!service.getMonthStartDate(false)).to.equal(false);
+      expect(!!service.getMonthStartDate({})).to.equal(false);
+      expect(!!service.getMonthStartDate({ uhc: false })).to.equal(false);
+    });
+
+    it('should return correct month_start_date', () => {
+      expect(service.getMonthStartDate({ uhc: { month_start_date: 3 }})).to.equal(3);
+      expect(service.getMonthStartDate({ uhc: { month_start_date: 3, visit_count: { month_start_date: 10 } }}))
+        .to.equal(3);
+      expect(service.getMonthStartDate({ uhc: { visit_count: { month_start_date: 10 } }})).to.equal(10);
+    });
+  });
+
+  describe('getVisitCountSettings', () => {
+    it('should return empty object for invalid input', () => {
+      expect(service.getVisitCountSettings(false)).to.deep.equal({});
+      expect(service.getVisitCountSettings({})).to.deep.equal({});
```

Comment:
It looks like you've removed some assertionts that made sure that the argument is optional. Maybe the code should be updated to set the argument as optional, right?